owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/CreateRemoteShareOperation.kt

```diff
@@ -0,0 +1,252 @@
+     */
+    private var mPublicUpload: Boolean?,
+    /**
+     * Password to set for the public link
+     */
+    private var mPassword: String?,
+    /**
+     * Access permissions for the file bound to the share
+     */
+    private var mPermissions: Int
+) : RemoteOperation<ShareParserResult>() {
+    private var mGetShareDetails: Boolean = false
+
+    /**
+     * Name to set for the public link
+     */
+    private var mName = ""
+
+    /**
+     * Expiration date to set for the public link
+     */
+    private var mExpirationDateInMillis: Long = 0
+
+    init {
+        mGetShareDetails = false        // defaults to false for backwards compatibility
+    }
+
+    /**
+     * Set name to create in Share resource. Ignored by servers previous to version 10.0.0
+     *
+     * @param name Name to set to the target share.
+     * Null or empty string result in no value set for the name.
+     */
+    fun setName(name: String?) {
+        this.mName = name ?: ""
+    }
+
+    /**
+     * Set password to create in Share resource.
+     *
+     * @param password Password to set to the target share.
+     * Null or empty string result in no value set for the password.
+     */
+    fun setPassword(password: String) {
+        mPassword = password
+    }
+
+    /**
+     * Set expiration date to create in Share resource.
+     *
+     * @param expirationDateInMillis Expiration date to set to the target share.
+     * Zero or negative value results in no value sent for expiration date.
+     */
+    fun setExpirationDate(expirationDateInMillis: Long) {
+        mExpirationDateInMillis = expirationDateInMillis
+    }
+
+    /**
+     * Set permissions to create in Share resource.
+     *
+     * @param permissions Permissions to set to the target share.
+     * Values <= 0 result in value set  to the permissions.
+     */
+    fun setPermissions(permissions: Int) {
+        mPermissions = permissions
+    }
+
+    /**
+     * * Enable upload permissions to create in Share resource.
+     * *
+     * * @param publicUpload  Upload permission to set to the target share.
+     * *                      Null results in no update applied to the upload permission.
+     */
+    fun setPublicUpload(publicUpload: Boolean?) {
+        mPublicUpload = publicUpload
+    }
+
+    fun setGetShareDetails(set: Boolean) {
+        mGetShareDetails = set
+    }
+
+    override fun run(client: OwnCloudClient): RemoteOperationResult<ShareParserResult> {
+        var result: RemoteOperationResult<ShareParserResult>
+
+        try {
+            val formBodyBuilder = FormBody.Builder()
+                .add(PARAM_PATH, mRemoteFilePath)
+                .add(PARAM_SHARE_TYPE, Integer.toString(mShareType.value))
+                .add(PARAM_SHARE_WITH, mShareWith)
+
+            if (mName.length > 0) {
+                formBodyBuilder.add(PARAM_NAME, mName)
+            }
+
+            if (mExpirationDateInMillis > 0) {
```

Comment:
use the constant you have created previously

Reply:
👍